File: semgrep-rules/python/cors/testdata/security_config.py
from django.views.decorators.csrf import csrf_exempt
from django.utils.decorators import method_decorator
from flask_cors import CORS
from markupsafe import escape
import jwt
import os
import pathlib
from werkzeug.utils import secure_filename


def insecure_cors(app):
    # ruleid: python-cors-wildcard-credentials
    CORS(app, origins="*", supports_credentials=True)


def default_wildcard_cors(app):
    # ruleid: python-cors-wildcard-credentials
    CORS(app, supports_credentials=True)


def safe_cors(app):
    # ok: python-cors-wildcard-credentials
    CORS(app, origins=["https://example.com"], supports_credentials=True)


@csrf_exempt
def exempt_view(request):
    # ruleid: python-django-csrf-exempt
    return request


@method_decorator(csrf_exempt, name="dispatch")
class ExemptView:
    pass


def disable_wtf_csrf(app):
    # ruleid: python-flask-wtf-csrf-disabled
    app.config["WTF_CSRF_ENABLED"] = False


def safe_wtf_csrf(app):
    # ok: python-flask-wtf-csrf-disabled
    app.config["WTF_CSRF_ENABLED"] = True


# ruleid: python-hardcoded-secret-key
SECRET_KEY = "super-secret"


def hardcoded_secret(app, payload, token):
    # ruleid: python-hardcoded-secret-key
    app.config["SECRET_KEY"] = "another-secret"
    # ruleid: python-hardcoded-secret-key
    jwt.encode(payload, "jwt-secret", algorithm="HS256")
    # ruleid: python-hardcoded-secret-key
    jwt.decode(token, "jwt-secret", algorithms=["HS256"])


def safe_secret(app, payload):
    # ok: python-hardcoded-secret-key
    app.config["SECRET_KEY"] = os.environ["SECRET_KEY"]
    return jwt.encode(payload, os.environ["JWT_SECRET"], algorithm="HS256")


def insecure_upload(file, upload_dir):
    # ruleid: python-upload-unsafe-filename-path
    file.save(os.path.join(upload_dir, file.filename))
    # ruleid: python-upload-unsafe-filename-path
    return pathlib.Path(upload_dir) / file.filename


def safe_upload(file, upload_dir):
    # ok: python-upload-unsafe-filename-path
    file.save(os.path.join(upload_dir, secure_filename(file.filename)))

File: semgrep-rules/go/cors/testdata/security_config.go
package testdata

import (
	"mime/multipart"
	"path/filepath"

	"github.com/gin-contrib/cors"
	"github.com/gin-gonic/gin"
	"github.com/golang-jwt/jwt/v5"
)

func insecureCors() {
	// ruleid: go-cors-wildcard-credentials
	_ = cors.Config{AllowAllOrigins: true, AllowCredentials: true}
}

func safeCors() {
	// ok: go-cors-wildcard-credentials
	_ = cors.Config{AllowOrigins: []string{"https://example.com"}, AllowCredentials: true}
}

func insecureSecret(token *jwt.Token) error {
	// ruleid: go-hardcoded-jwt-secret
	_, _ = jwt.Parse("raw-token", func(token *jwt.Token) (interface{}, error) { return []byte("super-secret"), nil })
	// ruleid: go-hardcoded-jwt-secret
	_, err := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{"sub": "1"}).SignedString([]byte("super-secret"))
	return err
}

func safeSecret(secret []byte) error {
	// ok: go-hardcoded-jwt-secret
	_, err := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{"sub": "1"}).SignedString(secret)
	return err
}

func insecureUpload(c *gin.Context, f *multipart.FileHeader, dir string) error {
	// ruleid: go-upload-filename-path
	return c.SaveUploadedFile(f, filepath.Join(dir, f.Filename))
}

func safeUpload(c *gin.Context, f *multipart.FileHeader, dir string) error {
	// ok: go-upload-filename-path
	return c.SaveUploadedFile(f, filepath.Join(dir, filepath.Base(f.Filename)))
}
